feat: figure out sender for radix

kamiyaa opened the pull request:
### Description

Parse manifest and figure out who paid for this transaction.
Use/assume the payer as the sender of the transaction.

### Testing

added unittests to make sure parsing manifest is passing

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Transaction lookups now parse Radix manifests to identify the fee payer as the sender.
  * Transaction details include receipt metrics (gas used, cumulative gas used, effective gas price).
  * Added utility to convert Radix internal addresses to H256 for consistent address handling.

* **Tests**
  * Added manifest-based tests covering multiple Radix networks and scenarios (lock fee, withdraw, blob publish).

* **Chores**
  * Added a workspace-scoped dev-dependency to improve test tracing.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kamiyaa commented on rust/main/chains/hyperlane-radix/src/provider/radix.rs:
```diff
@@ -636,6 +638,20 @@ impl HyperlaneProvider for RadixProvider {
+            );
+        };
+
+        let blob_provider = BlobProvider::new();
```
not sure where we get this, so I just created a default one

yjamin replied:
Yeah, I'm also not sure what the performance is on manifest::compile

yjamin commented on rust/main/chains/hyperlane-radix/src/manifest.rs:
```diff
@@ -0,0 +1,192 @@
+    network: &NetworkDefinition,
+    blobs: impl IsBlobProvider,
+) -> Option<H256> {
+    let val = match manifest::compile(s, network, blobs) {
```
Manifest can include blobs, can you maybe check whether or not this still works if you compile a manifest that references a blob? Because you provide an empty blob provider

kamiyaa commented:
> Please replace the `manifest_with_blobs.txt` with the following:
> 
> ```
> CALL_METHOD
>     Address("account_tdx_2_129gthzx2r70rdylg5qxpkx7lpmaqaqtex6xca2a9cfnqzpgn5av5z8")
>     "lock_fee"
>     Decimal("169.618337617485")
> ;
> PUBLISH_PACKAGE_ADVANCED
>     Enum<0u8>()
>     None
>     Blob("404926490ca92c49cf024b66f0a17c4c05d222fa3800b404bca32af0b9308016")
>     Map<String, Tuple>()
>     Enum<0u8>()
> ;
> ```

Updated and changed extension to rtm

yjamin commented on rust/main/chains/hyperlane-radix/src/provider/radix.rs:
```diff
@@ -641,6 +642,16 @@ impl HyperlaneProvider for RadixProvider {
+
+        // We assume the account that locked up XRD to pay for fees is the sender of the transaction
+        let fee_payer = find_fee_payer_from_manifest(&tx_manifest, &self.conf.network)
+            .ok_or_else(|| HyperlaneRadixError::ParsingError("fee_payer not found".to_owned()))?;
```
As we can't be sure who paid the fee its better to default to `Option::None` instead of throwing an error

kamiyaa replied:
Updated to use H256::zero()

yjamin commented on rust/main/chains/hyperlane-radix/src/provider/radix.rs:
```diff
@@ -673,10 +684,16 @@ impl HyperlaneProvider for RadixProvider {
+            // TODO: hard to tell what the tx interacted with, this can be with more than just one person, maybe use the the first component address?
+            recipient: None,
+            receipt: Some(TxnReceiptInfo {
+                gas_used: decimal_to_u256(fee_paid),
```
`gas_used` should be gas_limit and not the paid amount in XRD
